src/App.js

```diff
@@ -21,31 +21,50 @@ function useApp() {
   return { centralTeams, pacificTeams };
 }
 
+function SortTeamsByRank(teams, year) {
+  const teamsInTheYear = teams.map((team) => {
+    const result = team.results.find((el) => el.year === year);
+    return {
+      id: team.id,
+      name: team.name,
+      founded_year: team.founded_year,
+      league: team.league,
+      result: result,
+    };
+  });
+
+  return teamsInTheYear.sort(function (a, b) {
+    return a.result.rank - b.result.rank;
+  });
+}
+
+function Team({ team, key }) {
+  return (
+    <li>
+      {team.name}, {team.founded_year}, {team.result.rank}
+    </li>
+  );
+}
+
 function App() {
   const { centralTeams, pacificTeams } = useApp();
+  const sortedCentralTeams = SortTeamsByRank(centralTeams, 2020);
+  const sortedPacificTeams = SortTeamsByRank(pacificTeams, 2020);
```

Comment:
Though I'm not sure your intension, you can put `SortTeamsByRank` inside `useApp()`.
It depends on what you think `useApp`'s responsibility.

Reply:
Exactly...! 85b17ec69d74a0940ae1683a0d9537a5e5218337